cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/TranslatedStmt.qll

```diff
@@ -464,13 +569,25 @@ class TranslatedNoValueReturnStmt extends TranslatedReturnStmt, TranslatedVariab
     result = this.getEnclosingFunction().getReturnSuccessorInstruction(kind)
   }
 
+  final override TranslatedElement getChildInternal(int id) {
+    result = TranslatedVariableInitialization.super.getChildInternal(id)
+  }
+
+  override Instruction getChildSuccessorInternal(TranslatedElement child, EdgeKind kind) {
+    result = TranslatedVariableInitialization.super.getChildSuccessorInternal(child, kind)
+    or
+    result = TranslatedReturnStmt.super.getChildSuccessorInternal(child, kind)
+  }
+
   final override Type getTargetType() { result = this.getEnclosingFunction().getReturnType() }
 
   final override TranslatedInitialization getInitialization() { none() }
 
   final override IRVariable getIRVariable() {
     result = this.getEnclosingFunction().getReturnVariable()
   }
+
+  override predicate handlesDestructorsExplicitly() { any() }
```

Comment:
It seems like all overrides of `TranslatedReturnStmt` now override `handlesDestructorsExplicitly`. Could this be moved up to the base class as a final predicate?

Reply:
Done.